cmd/utils.go

```diff
@@ -1060,3 +1060,63 @@ func IsEmptyDir(name string) bool {
 
 	return err == io.EOF
 }
+
+func downloadFile(href string, writer io.Writer) error {
+
+	// allow file:// scheme
+	t := &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+	}
+	Debug.log("Proxy function for HTTP transport set to: ", &t.Proxy)
+	if runtime.GOOS == "windows" {
+		// For Windows, remove the root url. It seems to work fine with an empty string.
+		t.RegisterProtocol("file", http.NewFileTransport(http.Dir("")))
+	} else {
+		t.RegisterProtocol("file", http.NewFileTransport(http.Dir("/")))
+	}
+
+	httpClient := &http.Client{Transport: t}
+
+	req, err := http.NewRequest("GET", href, nil)
+	if err != nil {
+		return err
+	}
+
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
+		buf, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			Debug.log("Could not read contents of response body: ", err)
+		} else {
+			Debug.logf("Contents http response:\n%s", buf)
+		}
+		resp.Body.Close()
```

Comment:
Since we are moving this function, we could fix this bug. This is already performed by the `defer`.